lib/matplotlib/backend_bases.py

```diff
@@ -2791,6 +2791,12 @@ def mouse_move(self, event):
             except (ValueError, OverflowError):
                 pass
             else:
+                ax = event.inaxes
+                artists = ax.artists + ax.images + ax.lines
```

Comment:
and the order should be `ax.collections + ax.patches + ax.lines + ax.artists + ax.images` to match the order used for sorting out the rendering order  in `Axes.draw` (around https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/axes/_base.py#L1993 images are special cased around line 2030).